core-web/libs/portlets/edit-ema/portlet/src/lib/store/models.ts

```diff
@@ -0,0 +1,42 @@
+import { CurrentUser } from '@dotcms/dotcms-js';
+import { DotExperiment, DotLanguage, DotPageToolUrlParams } from '@dotcms/dotcms-models';
+import { InfoPage } from '@dotcms/ui';
+
+import { DotPageApiParams, DotPageApiResponse } from '../services/dot-page-api.service';
+import { UVE_STATUS } from '../shared/enums';
+import { DotDeviceWithIcon, DotPage, NavigationBarItem } from '../shared/models';
+
+export interface UVEState {
+    $isEnterprise: boolean;
+    $pageAPIResponse?: DotPageApiResponse;
+    $languages: DotLanguage[];
+    $currentUser?: CurrentUser;
+    $experiment?: DotExperiment;
+    $error?: number;
+    $params?: DotPageApiParams;
+    $status: UVE_STATUS;
+    $isTraditionalPage: boolean;
+    $canEditPage: boolean;
+    $pageIsLocked: boolean;
+}
+
+export interface EditorToolbarState {
+    $device?: DotDeviceWithIcon;
+    $socialMedia?: string;
+    $isEditState: boolean;
+}
```

Comment:
pretty sure we need an editor store.

Reply:
We could but I think that would be another refactor